File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = [0, 0, 0]
    
    answer1 = [1, 2, 3, 4, 5]
    answer2 = [2, 1, 2, 3, 2, 4, 2, 5]
    answer3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    for i in range(len(answers)):
        if answer1[i % 5] == answers[i]:
            answer[0] += 1
        if answer2[i % 8] == answers[i]:
            answer[1] += 1
        if answer3[i % 10] == answers[i]:
            answer[2] += 1
    
    answer = [i + 1 for i in range(3) if answer[i] == max(answer)]
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const answer = [0, 0, 0];
    
    const person1 = [1, 2, 3, 4, 5];
    const person2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const person3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    answers.forEach((item, i) => {
        if ((person1[i % person1.length]) === item) answer[0]++;
        if ((person2[i % person2.length]) === item) answer[1]++;
        if ((person3[i % person3.length]) === item) answer[2]++;
    })
    
    const maxAnswer = Math.max(...answer)
    const result = [];
    answer.forEach((a, i) => {
        if (a === maxAnswer) result.push(i + 1)
    })
    
    return result;
}